search-provider/eks-article-metadata-query-provider.c

```diff
@@ -0,0 +1,445 @@
+                                                      "title",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "content_type",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "thumbnail_uri",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "ekn_id",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "child_tags",
+                                                      G_VARIANT_TYPE_STRING_ARRAY);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "tags",
+                                                      G_VARIANT_TYPE_STRING_ARRAY);
+
+      g_variant_builder_close (&builder);
+    }
+
+  shards_strv = strv_from_shard_list ((GSList *) shards);
+
+  eks_article_metadata_complete_query (state->provider->skeleton,
+                                       state->invocation,
+                                       (const gchar * const *) shards_strv,
+                                       g_variant_builder_end (&builder));
+}
+
+static void
+append_construction_prop_from_string (const gchar *key,
+                                      const gchar *str,
+                                      GArray      *values_array,
+                                      GPtrArray   *props_array,
+                                      guint       *nprops)
+{
+  guint index = values_array->len;
+  GValue *value = NULL;
+
+  /* Need to be careful to set value *after* the array has been resized */
+  g_array_set_size (values_array, values_array->len + 1);
+  value = &(g_array_index (values_array, GValue, index));
+  g_value_init (value, G_TYPE_STRING);
+  g_value_set_string (value, str);
+  g_ptr_array_add (props_array, g_strdup (key));
+  ++(*nprops);
+}
+
+static void
+append_construction_prop_from_variant (const gchar *key,
+                                       GVariant    *variant,
+                                       GArray      *values_array,
+                                       GPtrArray   *props_array,
+                                       guint       *nprops)
```

Comment:
This parameter is not needed; you can just use the length of `props_array`

Reply:
I think I had this here as an outparam, but it might not be necessary anymore.